crates/dojo-project/src/migration/mod.rs

```diff
@@ -21,22 +20,24 @@ use self::world::{Class, Contract};
 #[derive(Debug, Default)]
 pub struct ContractMigration {
     pub deployed: bool,
-    // pub salt: FieldElement,
+    pub salt: FieldElement,
     pub contract: Contract,
     pub artifact_path: PathBuf,
+    // not to be confused with `compiled_class_hash` fields in `contract`: `local` and `remote`
+    pub class_hash: FieldElement,
+    pub contract_address: Option<FieldElement>,
 }
 
 #[derive(Debug, Default)]
 pub struct ClassMigration {
     pub declared: bool,
     pub class: Class,
     pub artifact_path: PathBuf,
+    // not to be confused with `compiled_class_hash` fields in `class`: `local` and `remote`
```

Comment:
hm why not to be confused? shouldn't this match the local hash? could you add a bit more info?

Reply:
yup my mistake it should be the same, was just a quick band aid... found the bug that causing the diff

Reply:
correction: TIL `compiled_class_hash` and `class_hash` are different. `compiled_class_hash` is for declaring the class but the latter is the actual class hash of the contract. So, need the `class_hash` field to for registering it to the world later.